esx_service/cli/vmdkops_admin.py

```diff
@@ -726,38 +731,382 @@ def get_version():
     except:
         return NOT_AVAILABLE
 
+def get_tenant_from_db(name):
+    try:
+        auth.connect_auth_db()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+    
+    error_info, tenant = auth._auth_mgr.get_tenant(name)
+    return error_info, tenant
+
+def create_tenant_in_db(name, description, default_datastore, default_privileges, vms, privileges):
+    try:
+        auth.connect_auth_db()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+
```

Comment:
Should have a validate_tenant_properties() or similar function to validate that the provided props for the tenant are .... valid. Like can the description be a large block of text or the privileges are actually valid and supported ones and that the datastore actually exists. Even name should be of some max. width allowed by the DB.

Perhaps caller must check for validity of datastore and privileges perhaps, just mentioning.